### PR TITLE
Add global arguments section to dump-docs command

### PR DESCRIPTION
**Pull request by tbantle22**

### CODE REVIEW

File: go/cmd/dolt/commands/dump_docs.go [hunk 1]
```diff
@@ -34,6 +34,33 @@ const (
 		"title: CLI\n" +
 		"---\n\n" +
 		"# CLI\n\n"
+
+	globalArgumentsSection = `
+## Global Arguments
+
+Dolt subcommands are in transition to using the flags listed below as global flags.
+The sql subcommand is currently the only command that uses these flags. All other commands will ignore them.
```
Comment (zachmu) [inline go/cmd/dolt/commands/dump_docs.go:42]: This is out of date, yeah?

Comment (zachmu) [reply go/cmd/dolt/commands/dump_docs.go:42]: Can we define a var for where this is used in the main command and reference that here?

File: go/cmd/dolt/cli/command.go [hunk 2]
```diff
@@ -309,6 +309,22 @@ func CheckUserNameAndEmail(config *env.DoltCliConfig) bool {
...
+			}
+		}
+
+		str += fmt.Sprintf("    %16s - %s\n", cmd.Name(), cmd.Description())
+	}
+
+	return str
+}
+
 func (hc SubCommandHandler) PrintUsage(commandStr string) {
```
Comment (fulghum) [inline go/cmd/dolt/cli/command.go:328]: (minor) Looks like this function could use the new new `GetUsage` to create the string before it prints it.

**Review by zachmu — APPROVED** [review]
LGTM, some suggestions

File: go/cmd/dolt/commands/dump_docs.go (hunk 1 above)
Comment (fulghum) [inline go/cmd/dolt/commands/dump_docs.go:42]: @macneale4 – any other updates in here?

**Review by fulghum — APPROVED** [review]
Looks good! Neil may have some more updates for the global args section based on the migration progress, but those can be updated later. 

Thanks for fixing this! 🙏

**Review by zachmu — APPROVED** [review]
LGTM

File: go/cmd/dolt/commands/dump_docs.go (hunk 1 above)
Comment (tbantle22) [reply go/cmd/dolt/commands/dump_docs.go]: Talked to neil and updated this copy

**Pull request merged**